optional-integration-analysis/merge-sce.R

```diff
@@ -0,0 +1,128 @@
+## This script takes in individual SCE objects and merges them into one SCE object
+## for a data integration analysis.
+
+## Set up ----------------------------------------------------------------------
+
+# Load libraries
+library(optparse)
+library(SingleCellExperiment)
+
+# Declare command line options
+option_list <- list(
+  make_option(
+    opt_str = c("--input_metadata_tsv"),
+    type = "character",
+    help = "Path to the TSV file containing the list of library IDs corresponding to the libraries being integrated, must have a `library_id` column."
+  ),
+  make_option(
+    opt_str = c("--input_directory"),
+    type = "character",
+    help = "File path to the directory storing the input SCE object RDS files."
+  ),
+  make_option(
+    opt_str = c("-o", "--output_sce_file"),
+    type = "character",
+    help = "Path to output RDS file, must end in .rds"
+  ),
+  optparse::make_option(
+    c("--project_root"),
+    type = "character",
+    default = NULL,
+    help = "Path to the root directory for the R project and where the `utils` folder lives."
+  ),
+  make_option(
+    opt_str = c("-t", "--threads"),
+    type = "integer",
+    default = 1,
+    help = "Number of multiprocessing threads to use"
+  )
+)
+
+# Read the arguments passed
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# If project root is not provided use here::here()
+if(is.null(opt$project_root)){
+  project_root <- here::here()
+} else {
+  project_root <- opt$project_root
+}
+
+# Source in set up functions
+source(file.path(project_root, "utils", "setup-functions.R"))
+
+# Check R version
+check_r_version()
+
+# Set up renv
+setup_renv(project_filepath = project_root)
+
+# Check that file extension for output file is correct
+if(!(stringr::str_ends(opt$output_sce_file, ".rds"))){
+  stop("output file name must end in .rds")
+}
+
+# Check that input metadata file is provided
+if(is.null(opt$input_metadata_tsv)){
+  stop("TSV file with the list of library IDs associated with the individual SCE objects to merge is missing.")
+} else {
+  # List of library ids
+  input_metadata <- readr::read_tsv(opt$input_metadata_tsv)
+  library_ids <- input_metadata$library_id
+}
+
+# Check that there is more than one library id associated with an SCE to be merged
+if(length(library_ids) == 1){
+  stop("Only 1 input file provided, no merging or integration will be performed for this group")
+}
+
+# List of SCE filepaths
+sce_files <- library_ids |> purrr::map(~ file.path(opt$input_directory, paste0(.x, "_processed.rds")))
```

Comment:
```suggestion
sce_files <- file.path(opt$input_directory, input_metadata$sce_filepath)
```
You shouldn't need the `purrr::map` here. 
But also as I'm looking at this, I think we want to require the file path as a column in the metadata and then pull the files directly from the metadata. The reason is because when downloading data from ScPCA, each processed file is going to be in its own sample folder, where as that isn't the case when going from the core workflow. So we either have to search the input directory recursively for the files, or provide the entire path as input. I think right now, the entire input makes sense.

Reply:
Good point 👍